core/src/main/java/amino/run/kernel/client/KernelClient.java

```diff
@@ -28,41 +38,256 @@
  * @author iyzhang
  */
 public class KernelClient {
+    private static final int MIN_METRIC_POLL_PERIOD_MS = 1000; /* Minimum metrics poll period */
+    private static final int MAX_METRIC_POLL_PERIOD_MS = 128000; /* Maximum metrics poll period */
```

Comment:
You still don't explain here in comments why these numbers were chosen.  They seem fairly arbitrary.
Put yourself in the position of someone coming after you to change the code.  They will ask themselves, "can I change this number, to what should I change it, and why".

Reply:
Added the comment in code. Please let me know if it is ok
![Selection_021](https://user-images.githubusercontent.com/35334869/63351166-5b803280-c37c-11e9-8314-cb7b316d57ca.png)